File: web3rpc/rpc-specs/code-samples/python/eth/transaction/estimateGas.py
from web3 import Web3
from web3py_ext import extend

host = "https://api.baobab.klaytn.net:8651"

callObject = {
    "from": "0x51239f87c33e95e3bdb72e31d06b5306bcec81cc",
    "to": "0x87ac99835e67168d4f9a40580f8f5c33550ba88b",
    "gas": "0x100000",
    "gasPrice": "0x5d21dba00",
    "value": "0x0",
    "input": "0x8ada066e"
}

w3 = Web3(Web3.HTTPProvider(host))
eth_response = w3.eth.estimate_gas(callObject)

print(eth_response)

File: web3rpc/rpc-specs/code-samples/javascript/eth/transaction/estimateGas.js
const { JsonRpcProvider } = require("@klaytn/ethers-ext");
(() => {
  const provider = new JsonRpcProvider("https://public-en-baobab.klaytn.net");
  const callObject = {
    from: "0x51239f87c33e95e3bdb72e31d06b5306bcec81cc",
    to: "0x87ac99835e67168d4f9a40580f8f5c33550ba88b",
    gas: "0x100000",
    gasPrice: "0x5d21dba00",
    value: "0x0",
    input: "0x8ada066e",
  };
  provider.eth
    .estimateGas(callObject, {}, (err, data, response) => {})
    .then((data) => {
      console.log(data);
    });
})();
